pulp_container/app/token_verification.py

```diff
@@ -99,23 +110,41 @@ def authenticate(self, request):
 
     def authenticate_header(self, request):
         """
-        Build a formatted authenticate string.
+        Initialize the Wwww-Authenticate header.
 
         For example, a created string will be the in following format:
         realm="http://localhost:123/token",service="docker.io",scope="repository:app:push"
-        and will be used in the "WWW-Authenticate" header.
+        and will be used in the header.
         """
         realm = settings.TOKEN_SERVER
-        repository_path, access_action = _access_scope(request)
-
         authenticate_string = f'{self.keyword} realm="{realm}",service="{request.get_host()}"'
 
-        if repository_path:
-            scope = f"repository:{repository_path}:{access_action}"
-            authenticate_string += f',scope="{scope}"'
+        try:
+            scope = ScopeFactory.from_request(request)
```

Comment:
Why the trouble of introducing three new classes just to juggle three strings?

I know this is about JWT tokens, but does that mean we need to code like it was Java? ;)